### PR TITLE
Add percentiles to histograms.

### PR DESCRIPTION
**Pull request by mateoatr**
Add the logic to API Two to get percentiles in a histogram.

### CODE REVIEW

File: AggregateMetrics/AggregateMetrics/Two/Implementation/HistogramImplementation.cs [hunk 1]
```diff
@@ -60,6 +125,18 @@ public MetricTelemetry GetValueAndReset()
                     metric.Min = curMinValue;
                     metric.Max = curMaxValue;
                 }
+                else if (this.shouldCalculatePercentiles)
+                {
+                    PercentileAggregations percentiles = CalculatePercentiles();
```
Comment (SergeyKanzhelev) [inline AggregateMetrics/AggregateMetrics/Two/Implementation/HistogramImplementation.cs]: you'll need to swap list with the empty one here with interlocked operation so the list wouldn't be updated while you are doing calculations

File: AggregateMetrics/AggregateMetrics/Two/TelemetryClientExtensions.cs [hunk 2]
```diff
@@ -72,7 +72,9 @@ public static IMeter Meter(this TelemetryClient telemetryClient, string name, Me
         /// <param name="name">Name of the histogram.</param>
         /// <param name="aggregations">Types of aggregations to perform.</param>
         /// <returns>Returns a histogram implementation.</returns>
-        public static IHistogram Histogram(this TelemetryClient telemetryClient, string name, HistogramAggregations aggregations = HistogramAggregations.Mean | HistogramAggregations.MinMax)
+        public static IHistogram Histogram(this TelemetryClient telemetryClient, string name, HistogramAggregations aggregations = HistogramAggregations.Mean | 
+            HistogramAggregations.MinMax |
+            HistogramAggregations.Percentiles)
```
Comment (SergeyKanzhelev) [inline AggregateMetrics/AggregateMetrics/Two/TelemetryClientExtensions.cs]: do not make is a default choice.

**Comment by SergeyKanzhelev** [comment]
Add a simple unit test on Histogram.

File: AggregateMetrics/AggregateMetrics/Two/HistogramAggregations.cs [hunk 3]
```diff
@@ -26,6 +26,6 @@ public enum HistogramAggregations
         /// <summary>
         /// Calculate 50th percentile.
```
Comment (dnduffy) [inline AggregateMetrics/AggregateMetrics/Two/HistogramAggregations.cs]: Does the comment about the 50th percentile still apply?

Comment (mateoatr) [inline AggregateMetrics/AggregateMetrics/Two/HistogramAggregations.cs]: It doesn't, removing it!

File: AggregateMetrics/AggregateMetrics/Two/PercentileAggregations.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.AggregateMetrics.Two
```
Comment (dnduffy) [inline AggregateMetrics/AggregateMetrics/Two/PercentileAggregations.cs]: nitpick: StyleCop compliance, particularly doc comments, would be great for all new code.

**Comment by dnduffy** [comment]
:shipit:

File: AggregateMetrics/AggregateMetrics/Two/PercentileAggregations.cs (hunk 4 above)
Comment (mateoatr) [inline AggregateMetrics/AggregateMetrics/Two/PercentileAggregations.cs]: I will surely do!

**Pull request merged**